towers-of-hanoi.js

```diff
@@ -0,0 +1,120 @@
+console.log("Let's start with 3 pegs and 5 discs. Access it using the 'game' object.")
+console.log("Start by making a move with:'game.moveDisc(from, to)' For example: 'game.moveDisc(1, 3)' will move the top disc from peg 1 to peg 3.")
+console.log("At any time, you can use the 'resetBoard()' method to restart the game with the current configuration.")
+console.log("Use more than 3 pegs to not mess everything up! It's pretty easy!")
+
+var startingBoard = function (numberOfPegs, numberOfDiscs, beginningPeg) {
+    this.numberOfPegs = numberOfPegs;
+    this.numberOfDiscs = numberOfDiscs;
+    this.beginningPeg = beginningPeg;
+    this.board = this.createBoard(numberOfPegs, numberOfDiscs, beginningPeg);
+}
+
+startingBoard.prototype.createBoard = function (numberOfPegs, numberOfDiscs, beginningPeg) {
+    var board = [];
+    var emptyPeg;
+    this.numberOfPegs = numberOfPegs;
+    this.numberOfDiscs = numberOfDiscs;
+    this.beginningPeg = beginningPeg;
+    if (numberOfPegs > 2 && numberOfDiscs > 0) {
+        for (var i = 0; i < numberOfPegs; i++) {
+            if (beginningPeg === i + 1) {
+                emptyPeg = [];
+                for (var j = 0; j < numberOfDiscs; j++) {
+                    emptyPeg.unshift(j + 1);
+                }
+                board.push(emptyPeg);
+            } else {
+                board.push([]);
+            }
+        }
+        return board;
+    } else {
+        console.log("This isn't a valid game! Sorry! Try again!")
+    }
+}
+
+
+startingBoard.prototype.moveDisc = function (from, to) {
+    var fromPeg = this.board[from - 1];
+    var toPeg = this.board[to - 1];
+    var disc1 = fromPeg.slice(-1)[0];
+    var disc2 = toPeg.slice(-1)[0];
+    
+    if (from === to) {
+        console.log("That's probably not going to do much.")
+    } else {
+        if (!disc1) {
+            return console.log("Starting peg has no disc! Please try another peg.")
+        } else if (!disc2 || disc1 < disc2) {
+            console.log("Great Move! (Maybe?) Current Board: ");
+            toPeg.push(fromPeg.pop())
+            this.printBoard();
+        } else {
+            return console.log("You're trying to put a larger disc on a smaller disc! Didn't you read the rules??.")
+        }
+        
+        if (this.winnerCheck()) {
+            console.log("Congratulations! You ended the world! Time to start over!");
+            console.log("To change the board use game.createBoard, if you would like to try with different values!");
+            this.resetBoard();
+        }
+
+    }
+    
+}
+
+startingBoard.prototype.printBoard = function () {
+    this.board.map(function (peg) {
+        var printedPeg = "--- ";
+       
+        peg.map(function (disc) {
+            printedPeg += (disc + " ");
+        })
+
+        console.log(printedPeg);
+    }) 
+}
+
+startingBoard.prototype.resetBoard = function () {
+    this.board = this.constructBoard(this.pegAmount, this.discAmount, this.originPeg);
```

Comment:
`constructBoard` is undefined and so `resetBoard` doesn't work - did you mean `createBoard`?